### DISCUSSION TITLE
Is there any way to get the current datetime in the rules?

### DISCUSSION THREAD

**Discussion by eliasc9** [pocketbase/pocketbase#192] [Q&A]
Hi, I love this project. 
I have a question:
Is there any way to get the current datetime in the rules?

The only thing that occurs to me is to create a current_datetime table, and do creates or updates, and take from there.
But it would be useful to have the rules, and to be able to do sums and comparisons.

**Comment by ganigeorgiev** [comment]
There is currently no support for that in the rules and filters. We can add a constant similar to the common `CURRENT_TIMESTAMP` SQL function, but I'm not sure whether this would be very useful.

When you say _"do sums and comparisons"_ do you mean aggregations (that's also not supported at the moment from the Admin UI)?

Could you elaborate more on your use case and how for example you would want to use the current datetime in the rules?

**Reply by eliasc9** [reply]
Sure, apologize for my vague proposal, here's a slightly more detailed one:

1 - **Add a constant `CURRENT_TIMESTAMP` (names proposal: `@now`, `now`)**
     Use case:  
        a. To limit some resources in the time, for example, a user paid for an account for 1 month.
             So, imagine a table with a suscriptionEndTime, now, you can add in rules like `suscriptionEndTime > @now` in the rules.
             Or you can hide the database table after a time by adding `1658498819000 > @now`
        b. To unlock some resource after a date
             So, imagine that we need to unlock a post view 8 minutes after was created, now, you can add in rules `created + 8000 < @now`

2 - Aggregations, yes, I meant the same thing, perfect, I didn't know that already exist.

Thank you very much!

**Reply by Lex-2008** [reply]
I see @ganigeorgiev created ticket #197 to add these, but note in your "b" example: `created + 8000 < @now` - AFAIK, filters currently don't support math operations (like `+`), so in order to unlock a post view 8 minutes after was created, you'll have to manually create a field "unlockAt".

**Reply by ganigeorgiev** [reply]
Thanks for the follow up.
1.a - I've created an issue - https://github.com/pocketbase/pocketbase/issues/197 for the `@now` constant. It is added in the roadmap, (somewhere after the Dart SDK and the import/export feature).

1.b - Simple math expression combined with fields and constants is an interesting idea, but I'll have to think a little more on it, because currently the filter grammar doesn't have definition for `+/-` since it is not context aware (it doesn't know beforehand whether the field contains a string, number, bool, date, etc.). So for now, this is not planned, but I'll keep it mind when updating the lexer.

2 - Aggregations such as count/sum,etc. on a filter/rule level for now too is not planned because usually such functions combine several fields (even from across different datasets) and will not be very useful if it is added only for a single field/column and this is why in general aggregations and reporting is left to the user to handle on their own when use PocketBase as framework via the internal DB abstraction (there will be more documentation in the future for this).

**Reply by eliasc9** [reply]
Cool! Thanks for this amazing project.

**Reply by octave2000** [reply]
Please we need a big documentation on rules please

**Reply by jimafisk** [reply]
Just wanted to follow up on **1.b - Simple math expressions**, is it possible to add or subtract time from the [@now identifier](https://pocketbase.io/docs/api-rules-and-filters#now-identifier)? Thanks!

**Reply by ganigeorgiev** [reply]
@jimafisk No, I have it somewhere buried in my todo but math expressions currently are not supported. 
_With the next release there will be several additional macros/constants available - `@todayStart`, `@todayEnd`, `@monthStart`, `@monthEnd`, `@yearStart`, `@yearEnd`._
